File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* list1, ListNode* list2) {
        if(list1 == nullptr && list2 == nullptr){
        return nullptr;
        }else if(list1 == nullptr){
            return list2;
        }else if(list2 == nullptr){
            return list1;
        }
        ListNode* head = new ListNode(0);
        ListNode* tail;
        tail = head;

        while(list1 != nullptr && list2 != nullptr){
            if(list1->val > list2->val){
                tail->next = list2;
                list2 = list2->next;
            }else{
                tail->next = list1;
                list1 = list1->next;
            }
            tail = tail->next;
        }
        if(list1 == nullptr){
            tail->next = list2;
        }
         if(list2 == nullptr){
            tail->next = list1;
        }
        return head->next;
    }
};

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        if not list1 and not list2:
            return None
        elif not list1 and list2:
            return list2
        elif not list2 and list1:
            return list1
        node = ListNode()
        if list1.val > list2.val:
            node.val = list2.val
            list2 = list2.next
        else:
            node.val = list1.val
            list1 = list1.next
        dummy = node
        while list1 and list2:
            if list1.val > list2.val:
                dummy.next = list2
                list2 = list2.next
            else:
                dummy.next = list1
                list1 = list1.next
            
            dummy = dummy.next
        
        if not list1 and not list2:
            return node
        elif not list1 and list2:
            dummy.next = list2
        elif not list2 and list1:
            dummy.next = list1
        return node



            